src/seedu/addressbook/commands/EditExamCommand.java

```diff
@@ -37,20 +39,42 @@
     public EditExamCommand() {
     }
 
-    public EditExamCommand(int targetVisibleIndex, Map<String, String> changedDetails) {
-        super(targetVisibleIndex);
+    public EditExamCommand(int targetVisibleIndex, Map<ExamField, String> changedDetails) throws IllegalValueException {
+        super(-1, targetVisibleIndex);
+        if (changedDetails.isEmpty()) {
+            throw new IllegalValueException(MESSAGE_NO_ARGS_FOUND + EditExamCommand.MESSAGE_USAGE);
+        }
+        if (changedDetails.containsKey(ExamField.isPrivate)) {
+            String privateStatus = changedDetails.get(ExamField.isPrivate);
+            if (!isPrivateValid(privateStatus)) {
+                throw new IllegalValueException(MESSAGE_USAGE);
+            }
+        }
         this.changedDetails = changedDetails;
```

Comment:
Same as previous about subclassing

Reply:
whats this related to about subclassing. Unless u mean as to why I'm checking here. The reason is this- Parser checks if the strings are strings, ints are ints and inputs them into a command. It should not check if the string is a valid string as a command need for example.
Since my command has Map<String, String> as input, the check only occurs at a lower level. in fact, I'm not sure if I should have this private string check in Command class or in Exam class

Reply:
What I meant was super(-1, targetVisibleIndex); feels like a bad way to code it as the -1 is a bit off-putting. If you create a new class of commands that contains the examIndex attribute instead, it will be cleaner as you don't have to explicitly pass-1 here